beacon-chain/p2p/subnets.go

```diff
@@ -31,59 +34,68 @@ func (s *Service) FindPeersWithSubnet(ctx context.Context, index uint64) (bool,
 		// return if discovery isn't set
 		return false, nil
 	}
+
+	topic += s.Encoding().ProtocolSuffix()
 	iterator := s.dv5Listener.RandomNodes()
-	nodes := enode.ReadNodes(iterator, lookupLimit)
-	exists := false
-	for _, node := range nodes {
+	iterator = enode.Filter(iterator, s.filterPeerForSubnet(index))
+
+	currNum := uint64(len(s.pubsub.ListPeers(topic)))
+	wg := new(sync.WaitGroup)
+	for {
 		if err := ctx.Err(); err != nil {
 			return false, err
 		}
-		if node.IP() == nil {
-			continue
+		if currNum >= threshold {
+			break
 		}
-		// do not look for nodes with no tcp port set
-		if err := node.Record().Load(enr.WithEntry("tcp", new(enr.TCP))); err != nil {
-			if !enr.IsNotFound(err) {
-				log.WithError(err).Debug("Could not retrieve tcp port")
+		nodes := enode.ReadNodes(iterator, peersRequiredInSubnetSearch)
+		for _, node := range nodes {
+			info, _, err := convertToAddrInfo(node)
+			if err != nil {
+				continue
 			}
-			continue
+			go func() {
+				wg.Add(1)
```

Comment:
```suggestion
			wg.Add(1)
			go func() {
```
I think you need to add this outside of the anonymous function.